pages/graphql.vue

```diff
@@ -0,0 +1,235 @@
+              <div v-for="field in queryFields" :key="field.name">
+                <gql-field :gqlField="field" />
+              </div>
+            </div>
+
+            <input
+              v-if="mutationFields.length > 0"
+              id="mutations-tab"
+              type="radio"
+              name="side"
+              checked="checked"
+            />
+            <label v-if="mutationFields.length > 0" for="mutations-tab"
+              >Mutations</label
+            >
+            <div v-if="mutationFields.length > 0" class="tab">
+              <div v-for="field in mutationFields" :key="field.name">
+                <gql-field :gqlField="field" />
+              </div>
+            </div>
+
+            <input
+              v-if="subscriptionFields.length > 0"
+              id="subscriptions-tab"
+              type="radio"
+              name="side"
+              checked="checked"
+            />
+            <label v-if="subscriptionFields.length > 0" for="subscriptions-tab"
+              >Subscriptions</label
+            >
+            <div v-if="subscriptionFields.length > 0" class="tab">
+              <div v-for="field in subscriptionFields" :key="field.name">
+                <gql-field :gqlField="field" />
+              </div>
+            </div>
+
+            <input
+              v-if="gqlTypes.length > 0"
+              id="gqltypes-tab"
+              type="radio"
+              name="side"
+              checked="checked"
+            />
+            <label v-if="gqlTypes.length > 0" for="gqltypes-tab">Types</label>
+            <div v-if="gqlTypes.length > 0" class="tab">
+              <div v-for="type in gqlTypes" :key="type.name">
+                <gql-type :gqlType="type" />
+              </div>
+            </div>
+          </section>
+        </pw-section>
+      </aside>
+    </div>
+  </div>
+</template>
+
+<style scoped lang="scss">
+.tab {
+  max-height: calc(100vh - 172px);
+  overflow: auto;
+}
+</style>
+
+<script>
+import axios from "axios";
+import * as gql from "graphql";
+import AceEditor from "../components/ace-editor";
+
+export default {
+  components: {
+    "pw-section": () => import("../components/section"),
+    "gql-field": () => import("../components/graphql/field"),
+    "gql-type": () => import("../components/graphql/type"),
+    Editor: AceEditor
+  },
+  data() {
+    return {
+      url: "https://rickandmortyapi.com/graphql",
+      schemaString: "",
+      queryFields: [],
+      mutationFields: [],
+      subscriptionFields: [],
+      gqlTypes: []
+    };
+  },
+  methods: {
+    getSchema() {
+      const startTime = Date.now();
+      this.schemaString = "Loading...";
+
+      // Start showing the loading bar as soon as possible.
+      // The nuxt axios module will hide it when the request is made.
+      this.$nuxt.$loading.start();
+
+      axios
+        .post(this.url, {
+          query: gql.getIntrospectionQuery()
+        })
+        .then(res => {
```

Comment:
Why not adopting the `async/await` approach?

Reply:
Yup. Sure `async/await` makes sense, I just did it out of habit.

Sure. I will refactor the code to use that.